e2e/e2e_test.go

```diff
@@ -0,0 +1,172 @@
+						t.Fatal(`Expected signature message to contain the transaction id`)
+					}
+					ethSignaturesCollected++
+				}
+
+				if msg.GetType() == validatorproto.TopicSubmissionType_EthTransaction {
+					//Verify that the eth transaction message has been submitted
+					if msg.GetTopicEthTransactionMessage().TransactionId != fromHederaTransactionID(transactionResponse.TransactionID) {
+						t.Fatal(`Expected ethereum transaction message to contain the transaction id`)
+					}
+					ethTransactionHash = msg.GetTopicEthTransactionMessage().GetEthTxHash()
+					ethTransMsgCollected++
+				}
+			},
+		)
+
+	// Wait for topic consensus messages to arrive
+	time.Sleep(60 * time.Second)
+
+	// Check that all the validators have submitted a message with authorisation signature
+	if ethSignaturesCollected != validatorsCount {
+		t.Fatal(`Expected the count of collected signatures to equal the number of validators`)
+	}
+
+	// Verify the exactly on eth transaction hash has been submitted
+	if ethTransMsgCollected != 1 {
+		t.Fatal(`Expected to submit exaclty 1 ethereum transaction in topic`)
+	}
+
+	success, err := ethClient.WaitForTransactionSuccess(common.HexToHash(ethTransactionHash))
+	// Verify that the eth transaction has been mined and succeeded
+	if success == false {
+		t.Fatal(`Expected to mine successfully the broadcasted ethereum transaction`)
+	}
+
+	// Get the wrapped hbar balance of the receiver after the transfer
+	whbarBalanceAfter, err := whbarInstance.BalanceOf(&bind.CallOpts{}, whbarReceiverAddress)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("WHBAR balance after transaction: [%s]\n", whbarBalanceAfter)
+
+	// Verify that the ethereum address hass recieved the exact transfer amount of WHBARs
+	if (whbarBalanceAfter.Int64() - whbarBalanceBefore.Int64()) != hedera.HbarFrom(hBarAmount, "hbar").AsTinybar() {
+		t.Fatal(`Expected to recieve the exact transfer amount of WHBAR`)
+	}
+
+}
+
+func sendTransactionToCustodialAccount(senderAccount hedera.AccountID, custodialAccount hedera.AccountID, memo string, hBarAmount float64, client *hedera.Client) (hedera.TransactionResponse, error) {
+	fmt.Println(fmt.Sprintf(`Sending [%v] Hbars through the Bridge. Transaction Memo: [%s]`, hBarAmount, memo))
+	res, _ := hedera.NewTransferTransaction().AddHbarSender(senderAccount, hedera.HbarFrom(hBarAmount, "hbar")).
+		AddHbarRecipient(custodialAccount, hedera.HbarFrom(hBarAmount, "hbar")).
+		SetTransactionMemo(memo).
+		Execute(client)
+	rec, err := res.GetReceipt(client)
+	fmt.Println(fmt.Sprintf(`TX broadcasted. ID [%s], Status: [%s]`, res.TransactionID, rec.Status))
+	time.Sleep(1 * time.Second)
+	return res, err
+}
+
+func initClient(accID hedera.AccountID, pK hedera.PrivateKey) *hedera.Client {
+	client := hedera.ClientForTestnet()
+	client.SetOperator(accID, pK)
+	return client
+}
+
+func fromHederaTransactionID(id hedera.TransactionID) string {
```

Comment:
There's an existing function called `FromHederaTransactionID` in `txid.go`

Reply:
Fixed this!